sensorprocess/sensorprocess.go

```diff
@@ -19,16 +20,48 @@ type Config struct {
 	CartoFacade cartofacade.Interface
 	IsOnline    bool
 
-	Lidar s.TimedLidar
-	IMU   s.TimedMovementSensor
+	Lidar          s.TimedLidar
+	MovementSensor s.TimedMovementSensor
 
 	Timeout         time.Duration
 	InternalTimeout time.Duration
 	Logger          logging.Logger
 }
 
+// getInitialMovementSensorReading gets the initial movement sensor reading.
+// It discards all movement sensor readings that were recorded before the first lidar reading.
+func (config *Config) getInitialMovementSensorReading(ctx context.Context,
+	lidarReading s.TimedLidarReadingResponse,
+) (s.TimedMovementSensorReadingResponse, error) {
+	if config.MovementSensor == nil || (!config.MovementSensor.Properties().IMUSupported &&
+		!config.MovementSensor.Properties().OdometerSupported) {
+		return s.TimedMovementSensorReadingResponse{}, errors.New("movement sensor is not supported")
+	}
+	for {
+		movementSensorReading, err := config.MovementSensor.TimedMovementSensorReading(ctx)
+		if err != nil {
+			return s.TimedMovementSensorReadingResponse{}, err
+		}
+
+		var readingTime time.Time
```

Comment:
[opt] We could define the readingTime without the if statements via soemthing like:

`readingTime := movementSensorReading.TimedOdometerResponse.ReadingTime.Add(movementSensorReading.TimedOdometerResponse.ReadingTime.Sub(movementSensorReading.TimedIMUResponse.ReadingTime)) `

since if only one sensor is defined, it will be time.Time{} and if both, this function should give us the max time.

Not as clear as you have it so up to you if you want to try for a single definition